s3bench.go

```diff
@@ -104,19 +104,33 @@ func main() {
 	if !*skipCleanup {
 		fmt.Println()
 		fmt.Printf("Cleaning up %d objects...", *numSamples)
-		numDeleted := 0
 		svc := s3.New(session.New(), cfg)
-		for i := 0; i < *numSamples; i++ {
-			params := &s3.DeleteObjectInput{
-				Bucket: aws.String(*bucketName),
-				Key:    aws.String(fmt.Sprintf("%s%d", *objectNamePrefix, i)),
-			}
-			_, err := svc.DeleteObject(params)
-			if err == nil {
-				numDeleted++
+
+		numDeleted := 0
+		//max that can be deleted via DeleteObjects()
+		var chunkSize = 1000
+
+		keyList := make([]*s3.ObjectIdentifier, 0, chunkSize)
+		for i := 0; i < *numSamples; i++{
+			bar := s3.ObjectIdentifier{
+				Key: aws.String(fmt.Sprintf("%s%d", *objectNamePrefix, i)),
+				}
+			keyList = append(keyList, &bar)
+			if len(keyList) == chunkSize || i == *numSamples-1 {
+				fmt.Printf("Deleting %d objects\n", len(keyList))
+				params := &s3.DeleteObjectsInput {
+					Bucket: aws.String(*bucketName),
+					Delete:   &s3.Delete{
+						Objects: keyList}}
+				_, err := svc.DeleteObjects(params)
+				if err == nil {
```

Comment:
This implies that the commit size might get exceeded. It might be better to clear the list outside this if statement. Consider what happens if we can only commit up to k*commitSize objects at a time. If we fail k times we will be failing all other deletes afterwards as well.

The code provides a report on how many things were were able to cleanup. It should be a good start for now.